src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DelegateCreationExpression.cs

```diff
@@ -21,6 +21,14 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                 return new BoundDelegateCreationExpression(node.Syntax, loweredReceiver, methodOpt: null, isExtensionMethod: false, type: node.Type);
             }
 
+            if (node.Argument.Kind == BoundKind.MethodGroup)
+            {
+                var mg = node.Argument as BoundMethodGroup;
```

Comment:
>node.Argument as BoundMethodGroup [](start = 25, length = 33)

Usage of an "as" operator in this manner violates our coding style. Casts that aren't expected to fail should use regular cast syntax. Otherwise, results of the cast should be checked for null before being dereferenced.